lib/brains/tests/test_customer.py

```diff
@@ -1,72 +1,88 @@
+from datetime import datetime
+
 from django.core.urlresolvers import reverse
 
+from braintree.customer import Customer
+from braintree.customer_gateway import CustomerGateway
+from braintree.error_result import ErrorResult
+from braintree.exceptions import NotFoundError
+from braintree.successful_result import SuccessfulResult
 from nose.tools import eq_
 
 from lib.brains.forms import BuyerForm
+from lib.brains.models import BraintreeBuyer
 from lib.brains.tests.base import BraintreeTest
 from lib.buyers.models import Buyer
 
 
+def customer(**kw):
+    customer = {
+        'id': 'customer-id',
+        'created_at': datetime.now(),
+        'updated_at': datetime.now()
+    }
+    customer.update(**kw)
+    return Customer(None, customer)
+
+
+def successful_customer(**kw):
+    return SuccessfulResult({'customer': customer(**kw)})
+
+
+def error():
+    return ErrorResult(None, {'errors': {}, 'message': ''})
+
+
 class TestCustomer(BraintreeTest):
+    gateways = {'customer': CustomerGateway}
 
     def setUp(self):
         super(TestCustomer, self).setUp()
         self.url = reverse('braintree:customer')
 
     def test_solitude_buyer_exists(self):
-        self.set_mocks(
-            ('GET', 'customers/minimal', 404, 'customer-minimal'),
-            ('POST', 'customers', 201, 'customer-minimal')
-        )
-        buyer = Buyer.objects.create(uuid='minimal')
-        res = self.client.post(self.url, data={'uuid': 'minimal'})
+        self.mocks['customer'].find.side_effect = NotFoundError
+        self.mocks['customer'].create.return_value = successful_customer()
+
+        buyer = Buyer.objects.create(uuid='customer-id')
+        BraintreeBuyer.objects.create(buyer=buyer)
+        res = self.client.post(self.url, data={'uuid': 'customer-id'})
         eq_(res.status_code, 201)
         eq_(Buyer.objects.count(), 1)
+        eq_(BraintreeBuyer.objects.count(), 1)
         eq_(res.json['resource_pk'], buyer.pk)
 
     def test_solitude_buyer_doesnt_exist(self):
-        self.set_mocks(
-            ('GET', 'customers/minimal', 200, 'customer-minimal')
-        )
-        res = self.client.post(self.url, data={'uuid': 'minimal'})
+        self.mocks['customer'].find.return_value = customer()
+
+        res = self.client.post(self.url, data={'uuid': 'customer-id'})
         eq_(res.status_code, 201)
         buyer = Buyer.objects.get()
-        eq_(buyer.uuid, 'minimal')
+        eq_(buyer.uuid, 'customer-id')
+        braintree_buyer = BraintreeBuyer.objects.get()
+        eq_(braintree_buyer.braintree_id, 'test-{0}'.format(buyer.pk))
```

Comment:
The code under test needs to post to the braintree API, right? It seems risky to rely on the base test case teardown to verify that. It might be good to put an explicit assertion here that the braintree create method was called with the right parameters.